### PR TITLE
allow resource and securitycontext overrides for git-sync service

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

Allows configurable  resources and security to optimise git sync service

## Related Issues

https://github.com/astronomer/issues/issues/5355

## Testing

added static test cases to validate resource (requests and limits) and security context

## Merging

cherry-pick to 1.7 release.

### CODE REVIEW

**Comment by pgvishnuram** [comment]
@danielhoherd - adding resource configuration to git sync relay - is it okay to have both resources to be same

File: templates/git-sync-relay/git-sync-relay-deployment.yaml [hunk 1]
```diff
@@ -84,6 +84,8 @@ spec:
               value: "false"
             {{- end }}
             {{- end }}
+          resources:
+            {{- toYaml .Values.gitSyncRelay.resources| nindent 12 }}
```
Comment (danielhoherd) [inline templates/git-sync-relay/git-sync-relay-deployment.yaml:88]:
```suggestion
            {{- toYaml .Values.gitSyncRelay.resources | nindent 12 }}
```

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Looking good. Need to add tests and finish the PR description.

**Comment by pgvishnuram** [comment]
Yes sure - i was waiting for configuration where to same resource section for both the containers - okay let me complete rest of the pending work on this PR

**Comment by danielhoherd** [comment]
> Yes sure - i was waiting for configuration where to same resource section for both the containers - okay let me complete rest of the pending work on this PR

@pgvishnuram good point. I think it's better to have them separate. It's doubtful that they would require the exact same resources.

**Review by danielhoherd — APPROVED** [review]
LGTM! I really wish we would've gone with `.Values.gitSyncRelay.gitSync.whatever` and `.Values.gitSyncRelay.gitDaemon.whatever` originally, but oh well. The design is out there being used so we can't change it now.

File: tests/chart_tests/test_git_sync_relay_deployment.py [hunk 2]
```diff
@@ -181,3 +181,52 @@ def test_gsr_deployment_without_ssh_credentials_and_known_hosts(self, kube_versi
             "--root=/git",
         ]
         assert c_by_name["git-daemon"]["livenessProbe"]
+
+    def test_gsr_deployment_with_resource_overrides(self, kube_version):
+        """Test that gitsync  deployment are configurable with custom resource limits."""
```
Comment (danielhoherd) [inline tests/chart_tests/test_git_sync_relay_deployment.py:186]:
```suggestion
        """Test that gitsync relay deployment are configurable with custom resource limits."""
```

File: tests/chart_tests/test_git_sync_relay_deployment.py [hunk 3]
```diff
@@ -181,3 +181,52 @@ def test_gsr_deployment_without_ssh_credentials_and_known_hosts(self, kube_versi
...
+        resources = {
+            "requests": {"cpu": 0.1, "memory": "512Mi"},
+            "limits": {"cpu": 0.1, "memory": "512Mi"},
```
Comment (danielhoherd) [inline tests/chart_tests/test_git_sync_relay_deployment.py:189]: We should set these to something obviously wrong. In other areas we've had to adjust default resources, and when test cases aren't obviously wrong, sometimes those future adjustments end up being identical to custom value test cases. Something like
```suggestion
            "requests": {"cpu": 99.5, "memory": "777Mi"},
            "limits": {"cpu": 99.6, "memory": "888Mi"},
```

File: tests/chart_tests/test_git_sync_relay_deployment.py [hunk 4]
```diff
@@ -181,3 +181,52 @@ def test_gsr_deployment_without_ssh_credentials_and_known_hosts(self, kube_versi
...
+            values=values,
+        )
+        assert len(docs) == 1
+        doc = docs[0]
+        assert doc["kind"] == "Deployment"
+        assert doc["apiVersion"] == "apps/v1"
+        assert doc["metadata"]["name"] == "release-name-git-sync-relay"
+        c_by_name = {
+            c["name"]: c for c in doc["spec"]["template"]["spec"]["containers"]
+        }
```
Comment (danielhoherd) [inline tests/chart_tests/test_git_sync_relay_deployment.py:211]: We should add the get_containers_by_name function to this repo so we're not using this code snip everywhere. https://github.com/astronomer/astronomer/blob/da36df6cc8/tests/__init__.py#L14-L32

Comment (pgvishnuram) [reply tests/chart_tests/test_git_sync_relay_deployment.py:211]: Yes thought the same

File: tests/chart_tests/test_git_sync_relay_deployment.py [hunk 5]
```diff
@@ -181,3 +181,52 @@ def test_gsr_deployment_without_ssh_credentials_and_known_hosts(self, kube_versi
...
+        assert c_by_name["git-sync"]["resources"] == resources
+        assert c_by_name["git-daemon"]["resources"] == resources
+
+    def test_gsr_deployment_with_securitycontext_overrides(self, kube_version):
+        """Test that gitsync  deployment are configurable with custom securitycontext."""
+        gsrsecuritycontext = {"runAsUser": 65533, "privileged": True}
```
Comment (danielhoherd) [inline tests/chart_tests/test_git_sync_relay_deployment.py:217]: As above, we should change this to be something obviously wrong.
```suggestion
        gsrsecuritycontext = {"runAsUser": 12345, "privileged": True}
```

File: tests/chart_tests/test_git_sync_relay_deployment.py (hunk 4 above)
Comment (pgvishnuram) [reply tests/chart_tests/test_git_sync_relay_deployment.py:211]: adding it

Comment (pgvishnuram) [reply tests/chart_tests/test_git_sync_relay_deployment.py]: took reference from astronomer/astronomer and replaced the changes here

**Review by danielhoherd — APPROVED** [review]

**Pull request merged**